File: December 2022/17 December 2022/Python/Break a number.py
class Solution:
    def waysToBreakNumber (self,N):
        return (((N+1)*(N+2))//2)%(10**9+7)

#{ 
 # Driver Code Starts
#Initial Template for Python 3

if __name__ == '__main__':
    t = int(input())
    for _ in range(t):
        n = int(input())
        
        ob = Solution()
        print(ob.waysToBreakNumber(n))
        
# } Driver Code Ends

File: December 2022/17 December 2022/Java/Break a number.java
//{ Driver Code Starts

import java.io.*;
import java.util.*;

class GFG
{
    public static void main(String args[])throws IOException
    {
        BufferedReader read = new BufferedReader(new InputStreamReader(System.in));
        int t = Integer.parseInt(read.readLine());
        while(t-- > 0)
        {
            int N = Integer.parseInt(read.readLine());

            Solution ob = new Solution();
            System.out.println(ob.waysToBreakNumber(N));
        }
    }
}
// } Driver Code Ends

class Solution{
    int waysToBreakNumber(int N){
        int mod = 1000000007;
        long ans = (N+1)%mod;
        ans = ((ans%mod)*((N+2)%mod))%mod;
        ans/=2;
        ans%=mod;
        return (int)ans;
    }
}
